### PR TITLE
Add external links for posts

### LINKED ISSUE #110: Support external links in PostList
### Description

It would be useful if you could include **external links** in a `PostList`. This is common for blogs that wish to keep lists of affiliate links (e.g. to medium posts etc) in addition to local content.

I think that it could work through some combination of:

- In a post, allow for an `external_link` parameter.
- Somewhere [around here](https://github.com/sunpy/ablog/blob/main/ablog/post.py#L526) check for the external link parameter, and if it exists, simply create an _external_ `ref` that points to the link rather than an internal ref.

Then the user workflow would be:

- Publish a blog post externally
- Create a page in your Sphinx site, use `ablog` directive or frontmatter to add `external_link` param.
- All other configuration of the page relative to other posts would be the same

If folks think this would be useful, and nobody thinks I'm missing some complexity that would make this harder than it looks, I could give it a shot!

### PR DESCRIPTION
**Pull request by choldgraf**
### Description

This PR adds support for external links for posts, by adding an `external_link` parameter to the `Post` directive and page-level metadata. If this parameter is given, then when post lists or collections are created, the link will point to `external_link` rather than using an internal ref to the page.

This is a common pattern in a few other static site generators, since people may want to link to posts in some external site occasionally (such as Medium, or another blog).

To Do

- [x] Confirm that the implementation looks OK
- [x] Confirm the docs look correct
- [x] Write tests as needed

Fixes #110

### CODE REVIEW

File: ablog/templates/catalog.html [hunk 1]
```diff
@@ -1,5 +1,12 @@
 {%- extends "page.html" %} {% block body %} {% for collection in catalog %} {%
 if collection %}
+{% macro postlink(post) %}
```
Comment (choldgraf) [inline ablog/templates/catalog.html:3]: I could not for the life of me figure out how to import this macro from one file to another, apologies for the duplication :-/

**Comment by nabobalis** [comment]
The implementation looks good to me.

File: ablog/templates/catalog.html (hunk 1 above)
Comment (nabobalis) [reply ablog/templates/catalog.html:3]: I don't actually know how/if it is possible to import another macro, so I have no issue with the duplication.

**Comment by choldgraf** [comment]
Hey all - it has been a while since I looked at this one, but if folks would find it useful I can try to make time to get it done. What do others think?

**Comment by nabobalis** [comment]
Hi @choldgraf, if you would prefer, I could take over and try to get this over the finish line?

**Comment by choldgraf** [comment]
I'm more than happy for that to happen - you will probably be a lot more effective at that than I am, since you have more context on the ablog infrastructure 🙏

**Comment by nabobalis** [comment]
Works nicely if you see https://ablog--112.org.readthedocs.build/blog/category/manual/

Now to decide on tests.

**Comment by nabobalis** [comment]
TODO: Work out why locations is not at the sidebar for the doc build.

**Comment by nabobalis** [comment]
Sorry I forgot about this.

I've added a test, I will backport this to the 0.10 branch and do a release today.

**Pull request merged**